connectors/sources/sharepoint_online.py

```diff
@@ -1203,6 +1230,7 @@ async def _user_access_control_doc(self, user):
 
         prefixed_mail = _prefix_email(email)
         prefixed_username = _prefix_user(username)
+        prefixed_user_id = _prefix_user_id(user.get("id"))
         id_ = email if email else username
```

Comment:
Wondering can we use the `user.get("id")` here, instead of email/username?

Reply:
As we may want to have cross-system identities in the near future, it's probably better to use the email as the id as this is searchable.

Reply:
++. The sharepointID isn't typically easy for a customer to look up or reason about. But everyone knows their email address. This is why we decided to use email as the key (ES _id) for SPO.